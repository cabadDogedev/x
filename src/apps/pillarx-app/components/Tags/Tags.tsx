// components
import Body from '../Typography/Body';

type TagsProps = {
    icon: string;
    tagText: string;
}
const Tags = ({ icon, tagText }: TagsProps) => {
	return (
		<div className='flex gap-2 items-center bg-[#312F3A] py-1.5 px-4 rounded-md mobile:px-3'>
			<img src={icon} />
			<Body>{tagText}</Body>
		</div>
	)
}

export default Tags;
