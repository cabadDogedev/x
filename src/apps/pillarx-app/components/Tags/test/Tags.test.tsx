import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import Tags from '../Tags';

describe('<Tags />', () => {
    const icon = 'https://example.com/icon.png';
    const tagText = 'This is a tag text';
  
    it('renders correctly and matches snapshot', () => {
      const tree = renderer
        .create(<Tags icon={icon} tagText={tagText} />)
        .toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  
    it('renders the icon with the correct src', () => {
      const tree = renderer.create(<Tags icon={icon} tagText={tagText} />).toJSON() as ReactTestRendererJSON;
      const iconProp = (tree.children?.find(child => 
        typeof child === 'object' && child.type === 'img') as ReactTestRendererJSON) || null;
  
      expect(iconProp).not.toBeNull();
      expect(iconProp.type).toBe('img');
      expect(iconProp.props.src).toBe(icon);
    });

    it('does not render the icon', () => {
      const tree = renderer.create(<Tags tagText={tagText} />).toJSON() as ReactTestRendererJSON;
      const iconProp = (tree.children?.find(child => 
        typeof child === 'object' && child.type === 'img') as ReactTestRendererJSON) || null;
  
      expect(iconProp).toBeNull();
    });
  
    it('renders the tag text correctly', () => {
      const tree = renderer.create(<Tags icon={icon} tagText={tagText} />).toJSON() as ReactTestRendererJSON;
      const textProp = (tree.children?.find(child => 
        typeof child === 'object' && child.type === 'p') as ReactTestRendererJSON);
  
      expect(textProp).not.toBeNull();
      expect(textProp.type).toBe('p');
      expect(textProp.children).toContain(tagText);
    });

    it('does not render the tag text', () => {
      const tree = renderer.create(<Tags icon={icon} />).toJSON() as ReactTestRendererJSON;
      const textProp = (tree.children?.find(child => 
        typeof child === 'object' && child.type === 'p') as ReactTestRendererJSON);
  
      expect(textProp).toBeUndefined();
    });
  
    it('applies the correct style', () => {
      const tree = renderer.create(<Tags icon={icon} tagText={tagText} />).toJSON() as ReactTestRendererJSON;
      expect(tree.type).toBe('div');
      expect(tree.props.className).toContain('flex');
      expect(tree.props.className).toContain('gap-2');
      expect(tree.props.className).toContain('items-center');
      expect(tree.props.className).toContain('bg-[#312F3A]');
      expect(tree.props.className).toContain('py-1.5');
      expect(tree.props.className).toContain('px-4');
      expect(tree.props.className).toContain('rounded-md');
      expect(tree.props.className).toContain('mobile:px-3');
    });
  });
