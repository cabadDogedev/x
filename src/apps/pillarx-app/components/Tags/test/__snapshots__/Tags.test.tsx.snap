// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> renders correctly and matches snapshot 1`] = `
<div
  className="flex gap-2 items-center bg-[#312F3A] py-1.5 px-4 rounded-md mobile:px-3"
>
  <img
    src="https://example.com/icon.png"
  />
  <p
    className="text-base font-medium undefined"
  >
    This is a tag text
  </p>
</div>
`;
